### PR TITLE
Add Reports Page

### LINKED ISSUE #240: Implement questions report system
Users need a way to report questions that have some sort of error. 
- [x] Create `Report` model
- [x] Create CRD endpoints

### PR DESCRIPTION
**Pull request by LordOfTheNeverThere**
This PR adds the Moderator Page that allows to see the Reports that have been made, along with each Report Relevant information.
It also fixes the Select UI problems we previously had.
Also adds 2 functions in utils, one for making array only have unique elements, another to retrieve what element appear the most times in an array.
Closes #240

### CODE REVIEW

File: frontend/src/components/report/ReportTableToolbar.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+		var uniqueReportIDs = reportIDs.filter(isUnique);
+		setReportIDs(uniqueReportIDs);
+
+		var questionIDs = props.rows.map((row) => row['question']);
+		var uniqueQuestionIDs = questionIDs.filter(isUnique);
+		setQuestionIDs(uniqueQuestionIDs);
+
+		var types = props.rows.map((row) => row['type']);
+		var uniqueTypes = types.filter(isUnique);
+		setTypes(uniqueTypes);
```
Comment (JeronimoMendes) [inline frontend/src/components/report/ReportTableToolbar.js:42]: This variables should be constants

File: frontend/src/pages/ReportsPage.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	};
+
+	const handleChangeRowsPerPage = (event) => {
+		setRowsPerPage(parseInt(event.target.value, 10));
+		setPage(0);
+	};
+
+	const search = (reportID, questionID, chosenType, bodyQuery) => {
+		var reports = originalRows;
+		console.log(reports);
```
Comment (JeronimoMendes) [inline frontend/src/pages/ReportsPage.js:87]: Remove

File: frontend/src/pages/ReportsPage.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+			if (chosenType !== 'none') {
+				reports = reports.filter((report) => chosenType === report['type']);
+			}
+			if (bodyQuery.length > 0) {
+				reports = reports
+					.filter((report) => report['body']) //filter only rows with a body !== null
+					.filter((report) => {
+						return report['body'].toLowerCase().includes(bodyQuery.toLowerCase());
+					}); //Search is case insentive
+				console.log(reports);
```
Comment (JeronimoMendes) [inline frontend/src/pages/ReportsPage.js:104]: Remove

**Review by JeronimoMendes — CHANGES_REQUESTED** [review]

**Comment by LordOfTheNeverThere** [comment]
@JeronimoMendes Just fixed the Problems

**Review by JeronimoMendes — APPROVED** [review]

**Pull request merged**